src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -172,35 +172,42 @@ public SubscriberMonitor basicConsume(String queue, DeliverCallback deliverCallb
      */
     @Deprecated(forRemoval = true)
     public void basicCancel(Channel channel, String consumerTag) {
-        retryManager.withRetry((RetryManager.RetryAction) channel1 -> channel1.basicCancel(consumerTag), null, () -> channel, livenessMonitor, readinessMonitor).exceptionally(th -> {
-            LOGGER.error("Can not cancel consumer with tag '{}'", consumerTag, th);
-            return null;
+        CompletableFuture<CompletableFuture<Void>> tmp = createRetryBuilder()
+                .setAction(channel1 -> channel1.basicCancel(consumerTag))
+                .setChannelCreator(() -> channel)
+                .setMetrics(managerMetrics)
+                .createWithAction();
+
+        tmp.thenAccept(future -> {
+            future.exceptionally(ex -> {
+                LOGGER.error("Can not cancel consumer with tag '{}'", consumerTag, ex);
+                return null;
+            });
         });
     }
 
     public void basicCancel(String consumerTag, HealthMetrics metrics) {
-        retryManager.withRetry((RetryManager.RetryAction) channel -> channel.basicCancel(consumerTag), metrics).exceptionally(th -> {
-            LOGGER.error("Can not cancel consumer with tag '{}'", consumerTag, th);
-            return null;
+        CompletableFuture<CompletableFuture<Void>> tmp = createRetryBuilder()
+                .setAction(channel1 -> channel1.basicCancel(consumerTag))
+                .setMetrics(metrics)
+                .createWithAction();
+
+        tmp.thenAccept(future -> {
+            future.exceptionally(ex -> {
+                LOGGER.error("Can not cancel consumer with tag '{}'", consumerTag, ex);
+                return null;
+            });
         });
     }
 
-    /**
-     * Create new channel and resender for this channel.
-     * @param metrics Metrics which will used for operations
-     * @return Resender with new channel.
-     * @see Resender
-     */
-    public Resender createResender(HealthMetrics metrics, Runnable unlockSenderFunction) {
-        return new Resender(this::createChannel, unlockSenderFunction, tasker, scheduler, connectionIsClosed, configuration, metrics.getLivenessMonitor(), metrics.getReadinessMonitor());
-    }
-
     @Override
     public void close() throws IllegalStateException {
         if (connectionIsClosed.getAndSet(true)) {
             return;
         }
 
+        tasker.shutdownNow();
```

Comment:
Please call tasker.shutdown() and tasker.awaitTermination(...) befor the shutdownNow()